### ISSUE TITLE
ApplicationError should accept other types of thrown objects

### ISSUE THREAD

**Issue by enyo** [dart-lang/rpc#65]
When my app throws (for example a `NoSuchMethodError`) the RPC library doesn't return properly but also throws:

```
Exception: type 'NoSuchMethodError' is not a subtype of type 'Exception' of 'e'.
```

This is because the [`ApplicationError` expects an `Exception` in the constructor](https://github.com/dart-lang/rpc/blob/master/lib/src/errors.dart#L34). It hink this can be changed to dynamic.

**Comment by FaisalAbid** [comment]
Good point, i will investigate this shortly

**Comment by inc-cloitsch** [comment]
Line has changed to [39](https://github.com/dart-lang/rpc/blob/ffdc8ee3ee777f3ea7972280fd2f0d96d6213b38/lib/src/errors.dart#L39),  but bug is still present.

Please allow Errors as well.

FYI the ApplicationError is instantiated [here](https://github.com/dart-lang/rpc/blob/ffdc8ee3ee777f3ea7972280fd2f0d96d6213b38/lib%2Fsrc%2Fconfig%2Fmethod.dart#L163)

**Comment by enyo** [comment]
This is closed by 2feb5c6591c72640c4b79d7ad71ca87ad88b5972 (which has not been released yet)
